### PR TITLE
feat: port `scc` tactic

### PR DESCRIPTION
**Pull request by Komyyy**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/SCC.lean [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  let ty : Q(Prop) ← whnf (← getMainTarget)
+  let ~q($p ↔ $q) := ty | throwError "goal is not the form `p ↔ q`"
+  closeMainGoal (← proveEqv p q)
+
+
+/-- `scc` uses the available equivalences and implications to prove a goal of the form `p ↔ q`.
+```lean
+example (p q r : Prop) (hpq : p → q) (hqr : q ↔ r) (hrp : r → p) : p ↔ r := by
+  scc
+``` -/
```
Comment (digama0) [inline Mathlib/Tactic/SCC.lean:296]:
````suggestion
```
-/
````

**Comment by digama0** [comment]
While I'm sorry to bring this up on an already ported tactic, I'm not sure we should have this tactic at all. It is significantly overcomplicated for the task (if you want to prove two propositions are equivalent you get better proofs just using BFS twice) and it has zero uses in mathlib. (It is used as a component in TFAE but it is also a poor fit in that case.) This kind of algorithm is more useful for `cc`, where there are more interesting relations than just a handful of equivalent propositions.

**Comment by Komyyy** [comment]
@digama0 The main purpose of porting this tactic is optimization of `tfae_finish`, please see #4154.

**Comment by digama0** [comment]
It seems the reason `tfae_finish` was slow was because it was misusing `solveByElim`. I've written a faster version which is still based on `solveByElim` at #4422.

**Pull request closed without merging**